flow/workflows/cdc_flow.go

```diff
@@ -159,16 +145,43 @@ func GetChildWorkflowID(
 }
 
 // CDCFlowWorkflowResult is the result of the PeerFlowWorkflow.
-type CDCFlowWorkflowResult = CDCFlowState
+type CDCFlowWorkflowResult = CDCFlowWorkflowState
+
+func (w *CDCFlowWorkflowExecution) signalHandler(state *CDCFlowWorkflowState, v shared.CDCFlowSignal) {
+	w.logger.Info("received signal - ", v)
+	if v == shared.ShutdownSignal {
+		w.logger.Info("received shutdown signal")
+		state.ActiveSignal = v
+	} else if v == shared.PauseSignal {
```

Comment:
So PauseSignal triggers flipping pause state? Seems like a source of flakiness compared to having explicit Pause/Unpause signal. But then you'd have to consider what to do when both signals are set. Are signals not able to carry boolean data?

Reply:
I agree that the current logic is a little flaky, but at the same time I can't think of an obvious way it will fail, since we only read a signal at a time and we only maintain one state variable. Temporal Signals are able to carry serializable data, but carrying a boolean value on the Pause signal isn't very clear IMO. 

We could also use explicit Pause and Unpause signals and ignore an illegal one [pausing when already paused] but this would require two RPC endpoints exposed. Another consideration is that the UI can prevent illegal state changes by reading the current state of the workflow, but our query layer has no such ability.

Reply:
In the minute while it's asleep one can imagine someone sending redundant pause signals